libexec/rtld-elf/cheri/cheri_reloc.c

```diff
@@ -67,6 +67,8 @@ process___cap_relocs(Obj_Entry* obj)
 	 */
 	void * __capability data_base = get_datasegment_cap(obj);
 	const void * __capability code_base = get_codesegment_cap(obj);
+	data_base = cheri_clearperm(data_base, CAP_RELOC_REMOVE_PERMS);
+	code_base = cheri_clearperm(code_base, CAP_RELOC_REMOVE_PERMS);
```

Comment:
Are there any uses of the get_foosegment_cap functions that *don't* want this? That would reduce the number of changes.

Reply:
I though there was some but upon further inspection there isn't any use that does not want to strip VMEM, so I cleaned this up now.
There is one leftover explicit clearperm in  `init_cap_from_fragment`, is there a reason why that function does not use the `make_code/data_cap` functions? Maybe because the reloc_self stuff would need some small refactoring?